frontend/app/routes/protected/person-case/first-name.tsx

```diff
@@ -1,45 +1,97 @@
-import { Form } from 'react-router';
-import type { RouteHandle } from 'react-router';
+import { data, useFetcher } from 'react-router';
+import type { ActionFunctionArgs, RouteHandle } from 'react-router';
 
 import { useTranslation } from 'react-i18next';
+import * as v from 'valibot';
 
+import type { Route, Info } from './+types/first-name';
+
+import { requireAuth } from '~/.server/utils/auth-utils';
+import { i18nRedirect } from '~/.server/utils/route-utils';
+import { firstNameSchema } from '~/.server/validation';
 import { Button } from '~/components/button';
+import { ButtonLink } from '~/components/button-link';
+import { ErrorSummary } from '~/components/error-summary';
 import { InputField } from '~/components/input-field';
 import { PageTitle } from '~/components/page-title';
+import { getFixedT } from '~/i18n-config.server';
 import { handle as parentHandle } from '~/routes/protected/layout';
+import { getLanguage } from '~/utils/i18n-utils';
 
 export const handle = {
   i18nNamespace: [...parentHandle.i18nNamespace, 'protected'],
 } as const satisfies RouteHandle;
 
-export default function FirstName() {
+export async function loader({ context, request }: Route.LoaderArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  return {
+    documentTitle: t('protected:person-case.page-title'),
+    sessionData: context.session.inPersonSINCase,
+  };
+}
+
+export function meta({ data }: { data: { documentTitle: string } }) {
```

Comment:
Use React Router [Type Safety](https://reactrouter.com/explanation/type-safety#type-safety).

```suggestion
export const meta: Route.MetaFunction = ({ data }) => {
```